components/resourceDetails/resourceSharing/agentAccess/agentAccessOptionsMenu/removeButton/index.jsx

```diff
@@ -141,7 +169,24 @@ export default function RemoveButton({
 
 RemoveButton.propTypes = {
   resourceIri: PropTypes.string.isRequired,
-  permission: PropTypes.shape().isRequired,
+  permission: PropTypes.shape({
+    type: PropTypes.string,
+    acl: PropTypes.shape({
+      read: PropTypes.bool,
+      write: PropTypes.bool,
+      append: PropTypes.bool,
+      control: PropTypes.bool,
+    }),
+    webId: PropTypes.string.isRequired,
+    alias: PropTypes.string,
+  }).isRequired,
+  profile: PropTypes.shape({
```

Comment:
Since the `profile` shape is reused - and `acl` as well - perhaps these could go into a shared proptypes shape constants file somewhere and be imported?